test/test_type_promotion.py

```diff
@@ -172,18 +172,19 @@ def test_mixed_type_backward(self, device):
                 func(x, y).sum().backward()
 
     def _get_test_tensor(self, device, dtype, remove_zeros=False):
-        shape = [20, 20, 20]
+        shape = [10, 10, 10]
         if dtype == torch.bool:
-            tensor = torch.randint(0, 2, shape, device=device, dtype=dtype)
+            tensor = torch.randint(int(remove_zeros), 2, shape, device=device, dtype=dtype)
```

Comment:
I found this really recently:

https://github.com/pytorch/pytorch/blob/3990e9d1ca23e39bfa65fe1d907ddb9f9dbf0919/test/test_torch.py#L14045-L14066

Could we reuse this? I imagine if we wanted to remove zeros we'd just use `fill_ones=True` or something.

Reply:
I'm a bit confused by that function: for half it always returns float, which would then have to be converted externally back to half for the test? I think I'd end up needing a wrapper around it so not sure it would save anything.

fill_ones gives all ones -- to confirm the math is correct I think it's better to have a random/variety of values.

Reply:
> I'm a bit confused by that function: for half it always returns float, which would then have to be converted externally back to half for the test? I think I'd end up needing a wrapper around it so not sure it would save anything.

A wrapper around it would save 10 LOC and consolidate the testing. I think it is valuable to just have "one way" to perform testing.

> fill_ones gives all ones -- to confirm the math is correct I think it's better to have a random/variety of values.

We could change add a `has_zeros` function to `_make_tensor`: I think that would make sense. The functions in test_torch that call it have `has_zeros` flags. Eventually if we add more integer kernels for other operators, `test_torch` would also benefit from this change.